spec-tests/src/test/java/org/candlepin/spec/hypervisors/HypervisorHeartbeatSpecTest.java

```diff
@@ -0,0 +1,90 @@
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.hypervisors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.candlepin.dto.api.client.v1.AsyncJobStatusDTO;
+import org.candlepin.dto.api.client.v1.ConsumerDTO;
+import org.candlepin.dto.api.client.v1.HypervisorIdDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.UserDTO;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.client.api.OwnerClient;
+import org.candlepin.spec.bootstrap.data.builder.ConsumerTypes;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.util.Map;
+
+@SpecTest
+public class HypervisorHeartbeatSpecTest {
+
+    private static ApiClient client;
+    private static OwnerClient ownerApi;
+
+
+    @BeforeAll
+    public static void beforeAll() {
+        client = ApiClients.admin();
+        ownerApi = client.owners();
+    }
+
+    @Test
+    public void shouldUpdateLastCheckinDateOfConsumersOfGivenReporter() {
+        HypervisorCheckInSpecTest.CommonTestData data = new HypervisorCheckInSpecTest.CommonTestData();
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        UserDTO user = UserUtil.createUser(client, owner);
+        ApiClient userClient = ApiClients.basic(user);
+        String reporterId = StringUtil.random("reporter");
+
+        ConsumerDTO hostConsumer = createHostConsumer(owner, userClient, data, reporterId);
+        ApiClient hostConsumerClient = ApiClients.ssl(hostConsumer);
+
+        AsyncJobStatusDTO job = userClient.hypervisors()
+            .hypervisorHeartbeatUpdate(owner.getKey(), reporterId);
+        if (job != null) {
+            AsyncJobStatusDTO status = client.jobs().waitForJob(job.getId());
+            assertEquals("FINISHED", status.getState());
+        }
+        assertTrue(job.getName().contains("Hypervisor Heartbeat Update"));
+        hostConsumer = hostConsumerClient.consumers().getConsumer(hostConsumer.getUuid());
+        assertFalse(hostConsumer.getLastCheckin().isBefore(OffsetDateTime.now().minusDays(1)));
```

Comment:
I think it would be better to validate that the hostConsumer's last checkin falls on the same current day instead of verifying that it is not before yesterday and it looks like that is how the original ruby test works as well. You should be able to do something like:
```
OffsetDateTime lastCheckinDay = hostConsumer.getLastCheckin().truncatedTo(ChronoUnit.DAYS);
OffsetDateTime today = OffsetDateTime.now().truncatedTo(ChronoUnit.DAYS);
assertThat(lastCheckinDay).isEqualTo(today);
```